apps/keyboard/js/views/swipeable_page_view.js

```diff
@@ -0,0 +1,322 @@
+'use strict';
+
+/* global LayoutPageView, BaseView, KeyView */
+
+(function(exports) {
+/**
+ * SwipeablePageView is used to render a page which is swipeable, e.g.
+ * for emoji layout.
+ */
+
+function EmojiView() {
+  BaseView.apply(this, arguments);
+}
+
+EmojiView.prototype = Object.create(BaseView.prototype);
+
+EmojiView.prototype.render = function() {
+  var contentNode = document.createElement('button');
+  if (this.options.classNames) {
+    contentNode.classList.add.apply(contentNode.classList,
+                                    this.options.classNames);
+  }
+
+  contentNode.innerHTML = this.options.outputChar;
+
+  this.element = contentNode;
+  this.viewManager.registerView(this.target, this);
+};
+
+function SwipeablePageView() {
+  LayoutPageView.apply(this, arguments);
+  this.sections = [];
+  this.currentSectionIndex = 0;
+
+  this.indicators = [];
+}
+
+SwipeablePageView.prototype = Object.create(LayoutPageView.prototype);
+
+SwipeablePageView.prototype.render = function() {
+  var layout = this.layout;
+  var content = document.createDocumentFragment();
+
+  var container = document.createElement('div');
+  if (this.options.classNames) {
+    container.classList.add.apply(container.classList, this.options.classNames);
+  }
+
+  if (layout.specificCssRule) {
+    container.classList.add(layout.layoutName);
+  }
+
+  var page = document.createElement('div');
+  page.classList.add('emoji-page');
+
+  page.addEventListener('touchstart', this);
+  this.page = page;
+
+  var emojiRange = this.layout.emojiRange;
+
+  if (!emojiRange instanceof Array) {
+    throw 'Should specify a range for emojis';
+  }
+
+  var start = this.layout.emojiRange[0];  // codepoint for 😀   U+1F600
+  var end = this.layout.emojiRange[1];    // codepoint for 🙏   U+1F64F
+
+  var threshold = 18;
+  var columnCount = 6;
+  var i = 0;
+  var section;
+
+  for (var c = start; c < end; c++) {
+    // Create another section
+    if (i % threshold === 0) {
+      section = document.createElement('div');
```

Comment:
You can shorten this function by creating another function inside `SwipeablePageView` or create a `SwipeablePageSectionView` (but I don't know how useful it would be).